flink/src/main/java/org/apache/iceberg/flink/source/RowDataRewriter.java

```diff
@@ -0,0 +1,193 @@
+ * under the License.
+ */
+
+package org.apache.iceberg.flink.source;
+
+import java.io.Serializable;
+import java.util.Collection;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.stream.Collectors;
+import org.apache.flink.api.common.functions.RichMapFunction;
+import org.apache.flink.api.java.DataSet;
+import org.apache.flink.configuration.Configuration;
+import org.apache.flink.table.data.RowData;
+import org.apache.flink.table.types.logical.RowType;
+import org.apache.iceberg.CombinedScanTask;
+import org.apache.iceberg.DataFile;
+import org.apache.iceberg.FileFormat;
+import org.apache.iceberg.PartitionSpec;
+import org.apache.iceberg.Schema;
+import org.apache.iceberg.Table;
+import org.apache.iceberg.TableProperties;
+import org.apache.iceberg.encryption.EncryptionManager;
+import org.apache.iceberg.flink.FlinkSchemaUtil;
+import org.apache.iceberg.flink.sink.RowDataTaskWriterFactory;
+import org.apache.iceberg.flink.sink.TaskWriterFactory;
+import org.apache.iceberg.io.FileIO;
+import org.apache.iceberg.io.LocationProvider;
+import org.apache.iceberg.io.TaskWriter;
+import org.apache.iceberg.relocated.com.google.common.collect.Lists;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import static org.apache.iceberg.TableProperties.DEFAULT_NAME_MAPPING;
+
+public class RowDataRewriter implements Serializable {
+
+  private static final Logger LOG = LoggerFactory.getLogger(RowDataRewriter.class);
+
+  private final Schema schema;
+  private final PartitionSpec spec;
+  private final Map<String, String> properties;
+  private final FileFormat format;
+  private final LocationProvider locations;
+  private final String nameMapping;
+  private final FileIO io;
+  private final boolean caseSensitive;
+  private final EncryptionManager encryptionManager;
+  private final Table table;
+
+  private final TaskWriterFactory<RowData> taskWriterFactory;
+
+  public RowDataRewriter(Table table, PartitionSpec spec, boolean caseSensitive,
+                         FileIO io, EncryptionManager encryptionManager) {
+    this.table = table;
+    this.schema = table.schema();
+    this.spec = spec;
+    this.locations = table.locationProvider();
+    this.properties = table.properties();
+    this.io = io;
+    this.encryptionManager = encryptionManager;
+
+    this.caseSensitive = caseSensitive;
+    this.nameMapping = table.properties().get(DEFAULT_NAME_MAPPING);
+
+    String formatString = table.properties().getOrDefault(
+        TableProperties.DEFAULT_FILE_FORMAT, TableProperties.DEFAULT_FILE_FORMAT_DEFAULT);
+    this.format = FileFormat.valueOf(formatString.toUpperCase(Locale.ENGLISH));
+
+    RowType flinkSchema = FlinkSchemaUtil.convert(table.schema());
+    taskWriterFactory = new RowDataTaskWriterFactory(table.schema(),
+        flinkSchema,
+        table.spec(),
+        table.locationProvider(),
+        table.io(),
+        table.encryption(),
+        Long.MAX_VALUE,
```

Comment:
The `targetFileSizeBytes` will always be unlimited ?  Pls set a configuration keys for it,  [here](https://github.com/apache/iceberg/pull/1083/files#r438890925) is the discussion we had for spark compaction.

Reply:
I have read this discussion, but my understanding is that the limit of targetFileSizeBytes is in the reader, not the writer. 
the writer of the spark is [here](https://github.com/apache/iceberg/blob/master/spark/src/main/java/org/apache/iceberg/spark/source/RowDataRewriter.java#L105)

```
    TaskWriter<InternalRow> writer;
    if (spec.fields().isEmpty()) {
      writer = new UnpartitionedWriter<>(spec, format, appenderFactory, fileFactory, io.value(), Long.MAX_VALUE);
    } else {
      writer = new SparkPartitionedWriter(spec, format, appenderFactory, fileFactory, io.value(), Long.MAX_VALUE,
          schema, structType);
    }
```

I set the targetFileSizeBytes in [RewriteDataFilesActionBase#targetSizeInBytes method ](https://github.com/zhangjun0x01/iceberg/blob/RewriteDataFilesAction/core/src/main/java/org/apache/iceberg/actions/RewriteDataFilesActionBase.java#L200)

Reply:
The conclusion should be [here](https://github.com/apache/iceberg/pull/1083/files#r439180961).  OK,  we only have to control the read split size and ignore the write target file size.